lib/msal-common/src/url/UrlString.ts

```diff
@@ -157,6 +157,17 @@ export class UrlString {
 
         return match[2];
     }
+
+    static getAbsoluteUrl(relativeUrl: string, baseUrl: string): string {
+        if (relativeUrl[0] === Constants.FORWARD_SLASH) {
```

Comment:
So this would only support relative urls that start with `/`? In theory we could also support relative redirect uris that start with a folder path, for example.

Reply:
Not sure I know what that would look like. Don't all `redirectUris` need to be in the format `https://domain.com/path` ? All we're trying to do here is make the `https://domain.com` part optional.

Reply:
All of these are technically relative urls:

- `/path/auth.html`
- `path`
- `./path`
- `path1/path2`
- `auth.html`

Reply:
Gotcha, guess I misunderstood your original comment. I thought about this but I feel like supporting some of those makes url validation a lot harder. Let's say someone passes in just `myApp.com` as a `redirectUri`. String matching logic intended to match something like `auth.html` as a relative path may turn the `redirectUri` into `https://myApp.com/myApp.com` instead of the intended `https://myApp.com`. I think it's safer and less prone to unexpected behavior to tell people if they want to use relative paths it needs to start with `/` (which I think is the most common anyway)

Reply:
Yeah, makes sense, and we can always add in enhanced relative uri logic later if needed.